### PR TITLE
Add clearCache feature to fastify.view

### PR DESCRIPTION
**Pull request by savearray2**
#### Checklist

- [x] run `npm run test`
- [x] tests are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Pull request for the following feature mentioned here: https://github.com/fastify/point-of-view/issues/201
Please let me know if there's anything you would like changed, as it's my first time contributing to fastify.

Thank you :)

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -25,6 +25,74 @@ test('fastify.view exist', t => {
...
+  })
+})
+
+test('fastify.view.clearCache clears cache', t => {
+  t.plan(10)
+  const templatesFolder = path.join(__dirname, '..', 'tmp')
+  try {
+    fs.mkdirSync(templatesFolder)
+  } catch {}
+  fs.writeFileSync(path.join(templatesFolder, 'cache_clear_test.ejs'), '<html><body><span>123</span></body></<html>')
```
Comment (mcollina) [inline test/test.js:52]: Can you please use the system temp folder instead?

File: test/test.js [hunk 2]
```diff
@@ -25,6 +25,74 @@ test('fastify.view exist', t => {
...
+    production: true
+  })
+
+  fastify.get('/view-cache-test', (req, reply) => {
+    reply.type('text/html; charset=utf-8').view('cache_clear_test')
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    t.ok(fastify.view.clearCache)
```
Comment (mcollina) [inline test/test.js:70]: this assertion is not needed

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by savearray2** [comment]
@mcollina 
Thank you for your review. I've made the changes you've requested.
Please let me know if there is anything else you would like updated.

Thanks again :)

File: README.md [hunk 3]
```diff
@@ -294,6 +294,12 @@ fastify.register(myViewRendererPlugin,   {
 })
 ```
 
+To forcefully clear cache when in production mode, call the ```view.clearCache()``` function.
```
Comment (Eomm) [inline README.md:297]:
```suggestion
To forcefully clear cache when in production mode, call the `view.clearCache()` function.
```

File: test/test.js [hunk 4]
```diff
@@ -25,6 +26,73 @@ test('fastify.view exist', t => {
...
+    t.error(err)
+    sget({
+      method: 'GET',
+      url: 'http://localhost:' + fastify.server.address().port + '/view-cache-test'
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.headers['content-length'], '' + body.length)
+      t.strictEqual(response.headers['content-type'], 'text/html; charset=utf-8')
+      fs.writeFileSync(path.join(templatesFolder, 'cache_clear_test.ejs'), '<html><body><span>456</span></body></<html>')
+      fastify.view.clearCache()
```
Comment (Eomm) [inline test/test.js:79]: I would add another request before the clear to check:

- call `/view-cache-test` and it loads the file
- call `/view-cache-test` and it hits the cache
- `clearCache`
- call `/view-cache-test` and it loads the file again

**Comment by savearray2** [comment]
@Eomm 

Thank you for your input :)
I've made the changes you requested.
Please let me know if there's anything else.

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**